src/audio/volume/volume_hifi3.c

```diff
@@ -72,11 +75,17 @@ static void vol_s24_to_s24_s32(struct comp_dev *dev, struct audio_stream *sink,
 			mult = AE_MULF32S_LL(volume, AE_SLAA32(in_sample, 8));
 
 			/* Multiplication of Q1.31 x Q1.31 gives Q1.63.
-			 * Now multiplication is Q8.16 x Q1.31, the result
-			 * is Q9.48. Need to shift right by one to get Q17.47
-			 * compatible format for round.
+			 * Now multiplication is Q8.16 x Q1.31 (IPC3) or Q1.23 x
+			 * Q1.31 (IPC4), the result is appropriately
+			 * Q9.48 (IPC3) or Q1.55 (IPC4). Need to shift right by
+			 * 1 (IPC3) or 8 (IPC4) to get Q17.47 compatible format
+			 * for round.
 			 */
+#if CONFIG_IPC_MAJOR_3
```

Comment:
@singalsu @marcinszkudlinski can we pass the shift value as a non constant variable ? I'm guessing we cant as this would simplify.  If not we could define this value as a macro and set it via the CONFIG.

Reply:
Sorry I missed the context in the above question.
```c
out_sample = AE_ROUND32F48SSYM(AE_SRAI64(mult, 1));
#else /* CONFIG_IPC_MAJOR_4 */
			out_sample = AE_ROUND32F48SSYM(AE_SRAI64(mult, 8));
```
I means I assume the 1 and 8 above must be const.

Reply:
The shift parameter is legacy from time when volume did format conversion. The function should be renamed to vol_s24() and hard code and optimize all shifts to minimum.